ethclient/simulated/backend.go

```diff
@@ -188,3 +189,9 @@ func (n *Backend) AdjustTime(adjustment time.Duration) error {
 func (n *Backend) Client() Client {
 	return n.client
 }
+
+// Signer returns the signer for signing or validating transactions in the
+// simulated chain.
+func (n *Backend) Signer() types.Signer {
```

Comment:
I think it's weird to add this, since this method will not be available when using the regular ethclient. So the app needs to create the signer for its target chain anyway.

What we could do at some point, is adding a `Signer` method to ethclient. That method could do some probing via RPC to figure out the applicable signer for the remote chain.

Reply:
Do you mind if I drop your last two commits @rjl493456442 ?